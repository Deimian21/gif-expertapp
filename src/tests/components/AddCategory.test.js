import '@testing-library/jest-dom';
const { shallow } = require("enzyme")
const { AddCategory } = require("../../Components/AddCategory")
import React from 'react';
describe('Pruebas con AddCategory', () => {
    const setCategory= jest.fn();
    let wrapper= shallow( <AddCategory  setCategory={setCategory}/>);
    beforeEach(()=>{
      jest.clearAllMocks();
      wrapper= shallow( <AddCategory  setCategory={setCategory}/>);
    });
    test('debe tener la categoria  ', () => {
         
         expect(wrapper).toMatchSnapshot();
    })
    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', {target:{ value } });
        expect( wrapper.find('p').text().trim()).toBe(value);
    })
    test('no debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategory).not.toHaveBeenCalled();
    })
    
    test('debe llamar el setCategorys y limpiar la caja de texto', () => {
        const value='Hola Mundo';
        wrapper.find('input').simulate('change', {target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategory).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
        
    })
    
    
})
