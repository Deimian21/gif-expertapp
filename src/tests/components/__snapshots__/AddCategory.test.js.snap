// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con AddCategory debe tener la categoria   1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
  </p>
  <h2>
     AÃ±adir componete
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
